src/AddIns/Uno.UI.MediaPlayer.Skia.Gtk/GTKMediaPlayer.cs

```diff
@@ -0,0 +1,505 @@
+			_ = Task.Run(() =>
+			{
+				// Required to be on a background thread
+				// to avoid deadlocks
+				_mediaPlayer.Stop();
+			});
+
+			_videoView.SetVisible(false);
+		}
+		else
+		{
+			if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+			{
+				this.Log().Debug("Unable to stop, the player is not ready yet");
+			}
+		}
+	}
+
+	public void Pause()
+	{
+		if (EnsureMediaPlayerAndVideoView())
+		{
+			if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+			{
+				this.Log().Debug("Pause");
+			}
+
+			_ = Task.Run(() =>
+			{
+				// Required to be on a background thread
+				// to avoid deadlocks
+				_mediaPlayer.Pause();
+			});
+
+			_videoView.SetVisible(true);
+		}
+		else
+		{
+			if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+			{
+				this.Log().Debug("Unable to pause, the player is not ready yet");
+			}
+		}
+	}
+
+	public void SetVolume(int volume)
+	{
+		if (EnsureMediaPlayerAndVideoView())
+		{
+			if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+			{
+				this.Log().Debug($"SetVolume ({volume})");
+			}
+
+			_mediaPlayer.Volume = volume;
+			_videoView.SetVisible(true);
+		}
+		else
+		{
+			if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+			{
+				this.Log().Debug("Unable to set the volume, the player is not ready yet");
+			}
+		}
+	}
+
+	public void Mute(bool IsMuted)
+	{
+		if (EnsureMediaPlayerAndVideoView())
+		{
+			if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+			{
+				this.Log().Debug($"Mute {IsMuted}");
+			}
+			_mediaPlayer.Mute = IsMuted;
+		}
+		else
+		{
+			if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+			{
+				this.Log().Debug("Unable to mute, the player is not ready yet");
+			}
+		}
+	}
+
+	private void GtkMediaPlayer_LayoutUpdated(object? sender, object e)
+		=> UpdateVideoStretch();
+
+	private void UpdateVideoStretch()
+	{
+		_ = Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
+		{
+			if (_videoView != null &&
+					_mediaPlayer != null &&
+					_mediaPlayer.Media != null &&
+					_mediaPlayer.Media.Mrl != null &&
+					_videoContainer is not null)
+			{
+				var currentSize = new Size(ActualWidth, ActualHeight);
+
+				if (currentSize.Height <= 0 || currentSize.Width <= 0)
+				{
+					if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+					{
+						this.Log().Debug($"Skipping layout update for empty layout slot");
+					}
+
+					return;
+				}
+
+				var playerHeight = (double)currentSize.Height - _transportControlsBounds.Height;
+				var playerWidth = (double)currentSize.Width;
+
+				_mediaPlayer.Media.Parse(MediaParseOptions.ParseNetwork);
+
+				if (_mediaPlayer != null && _mediaPlayer.Media != null)
+				{
+					var videoTrack = _mediaPlayer.Media.Tracks.FirstOrDefault(track => track.TrackType == TrackType.Video);
+
+					if (_mediaPlayer.Media.Tracks.Any(track => track.TrackType == TrackType.Video))
```

Comment:
My previous comment here has not been addressed yet. There is something that we need to understand and fix / workaround https://github.com/unoplatform/uno/pull/12198#discussion_r1197935642

Reply:
Will address separately